### DISCUSSION TITLE
Very beginner question: polygons

### DISCUSSION THREAD

**Discussion by AveryAndrews** [opengisch/QField#2762] [Q&A]
I'm having a problem with polygon features, using QField 2.014 - Dev off the play store, and 3.22.4-Białowieża on Win10 PC, which us that when I add a polygon, hit the check, get the 'Add features' dialog and fill in the values for the field(s), and hit the check, the added polygon disappears about 2/10sec after the dialog closes.  Likewise, when the project is opened, the added polygons are present for a fraction of a second, and then disappear.  They also appear when 'show features list' is selected, but disappear when it is dismissed.  Is this the expected behavior?  If so, can it be suppressed?  No such problem occurs with point features, which remain on the screen after being saved. 

I've reinstalled the app, and generated a different version of the project with both points and polygons; the points behave as expected, the polygons continue to be invisible, regardless of whether the 'show on map' checkbox is ticked. Indeed, polygon behavior is similar to the point behavior when the 'show on map' checkbox is not ticked.

Update: seeing no answer, I've submitted a bug report.  The problem does not arise with shapefile polygon layers: they work as you would expect them to.

**Comment by suricactus** [comment]
@AveryAndrews providing a test project **always** greatly increases your chances to get an answer. Please attach a test project to see your.

I see your question was already answered here: https://github.com/opengisch/QField/issues/2800 . Respond to all places where you have asked a question so other users can benefit from your experience.

> @AveryAndrews 
> And today, things seem to work properly, on new projects created by both 2.22 and 2.24, so it will take some time to track down what the problem, presumably some kind of user error, is with the test projects where things don't work right.

**Comment by AveryAndrews** [comment]
Here is a sample project, created on 2.22, that shows the problem with the two layers with polygon geometry.  Perhaps somebody can figure out what is causing it.

[polytest.zip](https://github.com/opengisch/QField/files/8562284/polytest.zip)

**Reply by suricactus** [reply]
As Mathieu already answered, please be aware of layer ordering.
